sklearn/pipeline.py

```diff
@@ -217,122 +326,144 @@ def fit_predict(self, X, y=None, **fit_params):
-        final estimator. Valid only if the final estimator implements
-        transform.
+        """Apply transforms, and transform with the final estimator
 
         Parameters
         ----------
-        X : iterable
-            Data to predict on. Must fulfill input requirements of first step
-            of the pipeline.
+        X : array-like, shape = [n_samples, n_features]
+            Data samples, where ``n_samples`` is the number of samples and
+            ``n_features`` is the number of features.
+
+        Returns
+        -------
+        Xt : array-like, shape = [n_samples, n_transformed_features]
         """
         Xt = X
         for name, transform in self.steps:
-            Xt = transform.transform(Xt)
+            if transform is not None:
+                Xt = transform.transform(Xt)
         return Xt
 
     @if_delegate_has_method(delegate='_final_estimator')
     def inverse_transform(self, X):
-        """Applies inverse transform to the data.
-        Starts with the last step of the pipeline and applies
-        ``inverse_transform`` in inverse order of the pipeline steps.
-        Valid only if all steps of the pipeline implement inverse_transform.
+        """Apply inverse transformations in reverse order
 
         Parameters
         ----------
-        X : iterable
-            Data to inverse transform. Must fulfill output requirements of the
-            last step of the pipeline.
+        Xt : array-like, shape = [n_samples, n_transformed_features]
+            Data samples, where ``n_samples`` is the number of samples and
+            ``n_features`` is the number of features.
+
+        Returns
+        -------
+        Xt : array-like, shape = [n_samples, n_features]
         """
-        if X.ndim == 1:
+        if hasattr(X, 'ndim') and X.ndim == 1:
             warn("From version 0.19, a 1d X will not be reshaped in"
                  " pipeline.inverse_transform any more.", FutureWarning)
             X = X[None, :]
         Xt = X
-        for name, step in self.steps[::-1]:
-            Xt = step.inverse_transform(Xt)
+        for name, transform in self.steps[::-1]:
+            if transform is not None:
+                Xt = transform.inverse_transform(Xt)
         return Xt
 
     @if_delegate_has_method(delegate='_final_estimator')
     def score(self, X, y=None):
-        """Applies transforms to the data, and the score method of the
-        final estimator. Valid only if the final estimator implements
-        score.
+        """Apply transforms, and score with the final estimator
 
         Parameters
         ----------
-        X : iterable
-            Data to score. Must fulfill input requirements of first step of the
-            pipeline.
+        X : array-like, shape = [n_samples, n_features]
+            Evaluation data, where ``n_samples`` is the number of samples and
+            ``n_features`` is the number of features.
```

Comment:
Wherever all those things go in hasty rebases... I suspect none of these parameter descriptions were there when I first wrote the PR, and I should maintain some of the current docstrings...